internal/commands/pull/command.go

```diff
@@ -60,17 +68,36 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 		return err
 	}
 
+	templateZipPkg, templatePathTypes, err := cmd.inputs.resolveTemplate(ui, clients.Realm, app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
 	pathTarget, zipPkg, err := cmd.doExport(profile, clients.Realm, app.GroupID, app.ID)
 	if err != nil {
 		return err
 	}
 
+	var templatePathTargets []string
+	if templateZipPkg != nil {
+		for _, templatePathType := range templatePathTypes {
+			templatePathTarget := path.Join(pathTarget, templatePathType)
+			if proceed, err := checkPathDestination(ui, templatePathTarget); err != nil {
+				return err
+			} else if !proceed {
+				return nil
+			} else {
+				templatePathTargets = append(templatePathTargets, templatePathTarget)
+			}
+		}
+	}
+
 	pathRelative, err := filepath.Rel(profile.WorkingDirectory, pathTarget)
 	if err != nil {
 		return err
 	}
 
-	proceed, err := checkAppDestination(ui, pathTarget)
+	proceed, err := checkPathDestination(ui, pathTarget)
```

Comment:
No, since it'd make the directory for the `app` in making the template directories. 
I could move this above the template `checkPathDestination`s or only do this if `templateZipPkg` is nil

Reply:
I probably still don't have the full picture of what we expect/want to show the user if a template is involved vs when its not.  at least for when there's no template, we need to show this single prompt to tell them where we're writing contents.

by the sounds of it, it seems that when a template _is_ involved, we're pushing the "write logic" to potential 2 spots (`appDir/backend` and `appDir/frontend`) and so it would stand to reason with me that we'd want either:
  * a single confirm prompting the user about these two new spots that we're writing to
  * or two prompts confirming individually each spot

i think the question then becomes, do want to allow the user to ever partially write a template?  as in, if it had both frontend and backend paths, do you still want the user to have the ability to opt into one or both of them?  in other words, can a template still be "usable" if its only partially written ("partially" here being omitting either frontend, backend, or both)?  the answer to those questions likely drives what type of prompts we ultimately want to write for the user to see...

we should strive to keep the amount of expected prompts to a minimum (think "less button clicks" in the UI, make it streamlined).  so at the very least, we should not be prompting a third time here

Reply:
See @andrewzlchen 's comment below, but I was mostly following what was [written on the ticket](https://jira.mongodb.org/browse/REALMC-8257) in terms of where to put the templates